h5py/_hl/files.py

```diff
@@ -342,15 +343,20 @@ def __init__(self, name, mode=None, driver=None,
             with phil:
                 fid = h5i.get_file_id(name)
         else:
-            name = filename_encode(name)
+            if hasattr(name, 'read') and hasattr(name, 'seek'):
+                name = hex(id(name)).encode('ASCII')
+                driver = 'fileobj'
```

Comment:
Can you add a check there that the user did not pass in a non-default value for `driver` and raise if they did?

Reply:
Should that be a `ValueError` or `H5pyWarning`?

Reply:
Lean towards `TypeError` as that is what python gives you for invalid signatures.